components/form/UploadImageInput.tsx

```diff
@@ -54,19 +53,49 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
       const file = e.target.files[0];
 
       if (file) {
-        const imageName = uuidv4();
-        const imagePath = CDN + userId + '/' + imageName;
+        try {
+          const { data: listData, error: listError } = await supabase.storage
+            .from('images')
+            .list(userId, { recursive: true });
 
-        setImageSrc(imagePath);
-        setIsImageUploaded(true);
-        setError?.('');
+          if (listError) {
+            console.error('Error fetching old images:', listError);
+            return;
+          }
+
+          if (!listData) {
+            throw new Error('Failed to fetch images');
+          }
 
-        const { error } = await supabase.storage
-          .from('images')
-          .upload(userId + '/' + imageName, file);
+          const oldImages = listData.map((image) => image.name);
 
-        if (error) {
-          console.log(error);
+          if (oldImages.length > 0) {
+            const { error: deleteError } = await supabase.storage
+              .from('images')
+              .remove(oldImages.map((image) => userId + '/' + image));
```

Comment:
As I was going over this again to answer your question I spotted an issue that had gone overlooked in the first review - because you're now identifying the images with userId we are limiting all image uploads from that user to a single image - including for their items.

That means that if you upload a new profile picture you will replace the old one, which is what we wanted, but then if you upload a new item the image on that item will delete your profile picture as well

Reply:
@camelPhonso - I've fixed it by adding 'profile_' prefix to the name of the profile image file. At first, I wanted to create separate buckets for the items-images and profile-images. However, I always encountered the same error -` UploadImageInput.tsx:73 {statusCode: '403', error: 'Unauthorized', message: 'new row violates row-level security policy'}` which I think is because of the differences in my local and prod database configurations.

Let me know what you think and if it fixes the problem :)